File: basic_pkg/basic_pkg/basic_pub.py
import rclpy
from rclpy.node import Node
from rclpy.qos import QoSProfile
from std_msgs.msg import Int32


class BasicPublisher(Node):
    def __init__(self):
        super().__init__('basic_publisher')
        qos_profile = QoSProfile(depth=10)  # queue size
        self.publisher_ = self.create_publisher(Int32, 'topic', qos_profile)
        timer_period = 1  # sec
        self.timer = self.create_timer(timer_period, self.timer_callback)
        self.i = 0

    def timer_callback(self):
        msg = Int32(); msg.data = self.i
        self.publisher_.publish(msg)
        self.get_logger().warn('Published : %s'%msg.data)
        self.i += 1


def main(args=None):
    rclpy.init(args=args)
    pub_node = BasicPublisher()

    try:
        rclpy.spin(pub_node)
    except Exception as e:
        pub_node.get_logger().warn('Exception - %s'%e)
    finally:
        pub_node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()

File: basic_cpp_pkg/src/basic_pub.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int32.hpp"

using namespace std::chrono_literals;

class BasicCppPublisher : public rclcpp::Node
{
    public:
        BasicCppPublisher()
        : Node("basic_cpp_pub"), count_(0)
        {
            auto qos_profile = rclcpp::QoS(rclcpp::KeepLast(10));
            basic_publisher_ = this->create_publisher<std_msgs::msg::Int32>("topic", qos_profile);
            timer_ = this->create_wall_timer(1s, std::bind(&BasicCppPublisher::publish_msg, this));
        }
    
    private:
        void publish_msg()
        {
            auto msg = std_msgs::msg::Int32();
            msg.data = count_++;
            RCLCPP_WARN(this->get_logger(), "[BasicCppPub] Published : %d", msg.data);
            basic_publisher_->publish(msg);
        }
        rclcpp::TimerBase::SharedPtr timer_;
        rclcpp::Publisher<std_msgs::msg::Int32>::SharedPtr basic_publisher_;
        size_t count_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<BasicCppPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}